spec/kickstart/main.fmf

```diff
@@ -0,0 +1,71 @@
+story:
+    As a tester I want to specify detailed installation of a guest
+    using the kickstart script.
+
+description: |
+    As part of the :ref:`/spec/plans/provision` step it is
+    possible to use the ``kickstart`` key to specify additional
+    requirements for the installation of a guest. It is possible
+    to specify a kickstart script that will for example specify
+    specific partitioning.
+
+    Introduction
+    ^^^^^^^^^^^^
+
+    The structure of a kickstart file can be separated into several
+    sections.
+
+    The ``pre-install`` key corresponds to the `%pre` section of a file.
+    It can contain BASH commands, this part is run before
+    the installation of a guest.
+
+    The ``post-install`` key corresponds to the `%post` section of a file.
+    It can contain BASH commands, this part is after
+    the installation of a guest.
+
+    The ``script`` key contains the kickstart specific commands that
+    are run during the installation of a guest.
+
+    It is also possible to specify ``bkr-metadata`` that can be read by
+    beaker.
+
+
+    Notes
+    -----
+
+    The implementation for the ``kickstart`` key is in
+    progress. Support of a kickstart file is currently
+    limited to Beaker provisioning pool, that is
+    being  handled by Artemis.
+    Check individual plugin documentation for additional
+    information on the kickstart support.
+
+
+example:
+  - |
+    # Use the artemis plugin to provision a guest rom Beaker.
+    # The following `kickstart` specification will be run
+    # during the guest installation.
+    provision:
+        how: artemis
+        pool: beaker
+        image: rhel-7
+        kickstart:
+            pre-install: |
+                %pre --log=/dev/console
+                disk=$(lsblk | grep disk | awk '{print $1}')
+                echo $disk
+                %end
+            script: |
+                lang en_US.UTF-8
+                zerombr
+                clearpart --all --initlabel
+                part /boot --fstype="xfs" --size=200
+                part swap --fstype="swap" --size=4096
+                part / --fstype="xfs" --size=10000 --grow
+            post-install: |
+                %pre
+                systemctl disable firewalld
+                %end
+            bkr-metadata: |
```

Comment:
Hmm, I am not happy with the `bkr` in the  name. What about just calling this metadata. It is actually kickstart metadata anyway:

https://beaker-project.org/docs/user-guide/customizing-installation.html#kickstart-metadata

Reply:
yeah, if we can keep the hardware generic in terms of naming, if there is a good reason why not.

Reply:
I chnaged it to simply `metadata`